keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -203,7 +206,7 @@ def backbone_cls(cls):
     def preprocessor_cls(cls):
         return GPT2CausalLMPreprocessor
 
-    def call_with_cache(self, token_ids, padding_mask, cache, cache_index):
+    def call_with_cache(self, token_ids, cache, cache_index):
```

Comment:
Note that I am moving `padding_mask` out of the signature for now.

The only real use case I could think of is for left padded inputs, but that would only work if we switched to a gather op for the position embeddings, so we don't support that now anyway.

Seems best to start simple and add that in later with real support.

Reply:
Why would people pass a swiss cheese causal mask like that? Curious what the use case is.

Seems very niche to me to have specific tokens in the future with a fixed value.